dotnet/samples/Concepts/Agents/ComplexChat_NestedShopper.cs

```diff
@@ -0,0 +1,227 @@
+        """        
+        You are a personal shopper that provides gift ideas.
+
+        Only provide ideas when the following is known about the gift recipient:
+        - Relationship to giver
+        - Reason for gift
+
+        Request any missing information before providing ideas.
+
+        Only describe the gift by name.
+
+        Always immediately incorporate review feedback and provide an updated response.
+        """;
+
+    private const string ReviewerName = "ShopperReviewer";
+    private const string ReviewerInstructions =
+        """
+        Review the most recent shopping response.
+
+        Either provide critical feedback to improve the response without introducing new ideas or state that the response is adequate.
+        """;
+
+    private const string InnerSelectionInstructions =
+        $$$"""
+        Select which participant will take the next turn based on the conversation history.
+        
+        Only choose from these participants:
+        - {{{ShopperName}}}
+        - {{{ReviewerName}}}
+        - {{{LeaderName}}}
+        
+        Choose the next participant according to the action of the most recent participant:
+        - After user input, it is {{{ShopperName}}}'a turn.
+        - After {{{ShopperName}}} replies with ideas, it is {{{ReviewerName}}}'s turn.
+        - After {{{ShopperName}}} requests additional information, it is {{{LeaderName}}}'s turn.
+        - After {{{ReviewerName}}} provides feedback or instruction, it is {{{ShopperName}}}'s turn.
+        - After {{{ReviewerName}}} states the {{{ShopperName}}}'s response is adequate, it is {{{LeaderName}}}'s turn.
+                
+        Respond in JSON format.  The JSON schema can include only:
+        {
+            "name": "string (the name of the assistant selected for the next turn)",
+            "reason": "string (the reason for the participant was selected)"
+        }
+        
+        History:
+        {{${{{KernelFunctionSelectionStrategy.DefaultHistoryVariableName}}}}}
+        """;
+
+    private const string OuterTerminationInstructions =
+        $$$"""
+        Determine if user request has been fully answered.
+        
+        Respond in JSON format.  The JSON schema can include only:
+        {
+            "isAnswered": "bool (true if the user request has been fully answered)",
+            "reason": "string (the reason for your determination)"
+        }
+        
+        History:
+        {{${{{KernelFunctionTerminationStrategy.DefaultHistoryVariableName}}}}}
+        """;
+
+    [Fact]
+    public async Task RunAsync()
+    {
+        this.WriteLine($"! {Model}");
+
+        OpenAIPromptExecutionSettings jsonSettings = new() { ResponseFormat = ChatCompletionsResponseFormat.JsonObject };
+        OpenAIPromptExecutionSettings autoInvokeSettings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+
+        ChatCompletionAgent agentManager = CreateAgent(LeaderName, LeaderInstructions);
```

Comment:
Mixing terms here i.e. Manager -> Leader and above also use SpokesPerson

Reply:
Yes, I indeed struggled with that.  I'll do consistency pass now.

Reply:
Updated